presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -1259,6 +1261,46 @@ public void testMoney()
                 .execute(getQueryRunner(), postgresCreateAndInsert("tpch.presto_test_money"));
     }
 
+    @Test
+    public void testReal()
+    {
+        singlePrecisionFloatingPointTests(realDataType())
+                .execute(getQueryRunner(), prestoCreateAsSelect("presto_test_real"));
+
+        singlePrecisionFloatingPointTests(postgreSqlRealDataType())
+                .execute(getQueryRunner(), postgresCreateAndInsert("tpch.postgresql_test_real"));
+    }
+
+    @Test
+    public void testDouble()
+    {
+        doublePrecisionFloatinPointTests(doubleDataType())
+                .execute(getQueryRunner(), prestoCreateAsSelect("presto_test_double"));
+
+        doublePrecisionFloatinPointTests(postgreSqlDoubleDataType())
+                .execute(getQueryRunner(), postgresCreateAndInsert("tpch.postgresql_test_double"));
+    }
+
+    private static DataTypeTest singlePrecisionFloatingPointTests(DataType<Float> floatType)
+    {
+        return DataTypeTest.create()
+                .addRoundTrip(floatType, 123.45f)
```

Comment:
add a test exhausting precision of a float (~7 digits), to detect potential over-eager rounding